src/System.Net.Security/src/System/Net/Security/CipherSuitesPolicyPal.Linux.cs

```diff
@@ -0,0 +1,231 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Authentication;
+using System.Text;
+using Ssl = Interop.Ssl;
+using OpenSsl = Interop.OpenSsl;
+
+namespace System.Net.Security
+{
+    internal class CipherSuitesPolicyPal
+    {
+        private static readonly byte[] RequireEncryptionDefault =
+            Encoding.ASCII.GetBytes("DEFAULT\0");
+
+        private static readonly byte[] AllowNoEncryptionDefault =
+            Encoding.ASCII.GetBytes("ALL:eNULL\0");
+
+        private static readonly byte[] NoEncryptionDefault =
+            Encoding.ASCII.GetBytes("eNULL\0");
+
+        private byte[] _cipherSuites;
+        private byte[] _tls13CipherSuites;
+        private List<TlsCipherSuite> _tlsCipherSuites = new List<TlsCipherSuite>();
+
+        public IEnumerable<TlsCipherSuite> GetCipherSuites() => _tlsCipherSuites;
+
+        public CipherSuitesPolicyPal(IEnumerable<TlsCipherSuite> allowedCipherSuites)
```

Comment:
Since there's no interface here, just calling convention, I'd declare these members `internal` to make it clear that they aren't actually public.